components/service/glean/src/main/java/mozilla/components/service/glean/storages/GenericScalarStorageEngine.kt

```diff
@@ -79,22 +85,23 @@ internal abstract class GenericScalarStorageEngine<ScalarType> : StorageEngine {
     )
 
     /**
-     * Deserialize the metrics with a lifetime = User that are on disk.
+     * Deserialize the metrics with a particular lifetime that are on disk.
      * This will be called the first time a metric is used or before a snapshot is
      * taken.
      *
-     * @return A [SharedPreferences] reference that will be used to initialize [userLifetimeStorage]
+     * @param lifetime a [Lifetime] to deserialize
+     * @return A [SharedPreferences] reference that will be used to initialize [pingLifetimeStorage]
+     *         or [userLifetimeStorage] or null if an invalid lifetime is used.
      */
     @Suppress("TooGenericExceptionCaught")
-    open fun deserializeUserLifetime(): SharedPreferences {
-        val prefs =
-            applicationContext.getSharedPreferences(this.javaClass.canonicalName, Context.MODE_PRIVATE)
+    open fun deserializeLifetime(lifetime: Lifetime): SharedPreferences? {
```

Comment:
This function really _requires us_ to have `lifetime` either be `Lifetime.User || Lifetime.Ping`. We can tell Kotlin about that, by using the `require` function (see the [docs](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/require.html))